apps/api/src/domain/models/user-management/group/entities/coscrad-user-group.entity.ts

```diff
@@ -53,4 +59,30 @@ export class CoscradUserGroup extends Aggregate {
         // There are no complex invariant rules for a `CoscradUserGroup`
         return Valid;
     }
+
+    validateExternalState({
+        users,
+        userGroups: existingUserGroups,
+    }: InMemorySnapshot): InternalError | typeof Valid {
+        const allErrors: InternalError[] = [];
+        const existingUserIds = users.map(({ id }) => id);
```

Comment:
A helper function that makes this point free might be a nice addition to our 'functional' utils lib.